apps/browser-extension-wallet/src/hooks/useWalletManager.ts

```diff
@@ -205,6 +221,38 @@ export const connectHardwareWallet = async (model: Wallet.HardwareWallets): Prom
 const connectHardwareWalletRevamped = async (usbDevice: USBDevice): Promise<Wallet.HardwareWalletConnection> =>
   Wallet.connectDeviceRevamped(usbDevice);
 
+const deriveSharedWalletExtendedPublicKeyHash = async (
+  key: Crypto.Bip32PublicKeyHex,
+  derivationPath: KeyManagement.AccountKeyDerivationPath
+): Promise<Crypto.Ed25519KeyHashHex> => {
+  const accountKey = Crypto.Bip32PublicKey.fromHex(key);
+  const paymentKey = await accountKey.derive([derivationPath.role, derivationPath.index]);
+  return Crypto.Ed25519KeyHashHex(await paymentKey.hash());
+};
+
+const buildSharedWalletScript = async (
+  expectedSigners: Array<Crypto.Bip32PublicKeyHex>,
+  derivationPath: KeyManagement.AccountKeyDerivationPath
+) => {
+  const signers = [...expectedSigners].sort();
```

Comment:
sonarlint is complaining about this one
```suggestion
  const signers = [...expectedSigners].sort((key1, key2) => key1.localeCompare(key2)));
```